lib/rack/cors.rb

```diff
@@ -14,17 +14,17 @@ def initialize(app, opts={}, &block)
       @app = app
       @debug_mode = !!opts[:debug]
 
-      if logger = opts[:logger]
-        if logger.respond_to? :call
+      if opts[:logger]
+        if opts[:logger].respond_to? :call
           @logger_proc = opts[:logger]
         else
-          @logger = logger
+          @logger = opts[:logger]
         end
       end
 
       if block_given?
         if block.arity == 1
-          block.call(self)
+          yield self
         else
           instance_eval(&block)
```

Comment:
Instead of doing `&block` (using the `&block` from the method definition, which allocates unnecessary memory, see [above comment](https://github.com/cyu/rack-cors/pull/83/files#r41354723)), do `instance_eval &Proc.new`.

Calling `Proc.new` without any arguments in a method called with a block will return that block.